src/main/java/duke/models/tasks/TaskManager.java

```diff
@@ -60,18 +60,30 @@ public TaskManager() {
         return tasksWithThisDescription;
     }
 
-
     /**
      * Adds a Task to the list.
      *
-     * @param task The Task to be added to the list.
+     * @param taskToBeAdded contains the Task to be added to the list.
+     * @throws DukeException if the task is not added successfully.
      */
-    public void addTask(Task task) {
-        if (task.getId() == 0) {
+    public void addTask(Task taskToBeAdded) throws DukeException {
+        if (taskToBeAdded.getDescription().length() < 4) {
+            throw new DukeException("The task description is too short.");
+        }
+        ArrayList<Task> sameTasks = new ArrayList<>();
```

Comment:
You could probably replace this with a Boolean value like, `tasksAreTheSame`

Reply:
Actually I just realised that I don't really need to add this line.